### PR TITLE
Interactive authentication

### PR DESCRIPTION
**Pull request by pacc**
I had to implement this and it works,
it is not tested on multiple challenge logins.

### CODE REVIEW

File: sshtunnel.cpp [hunk 1]
```diff
@@ -84,11 +84,20 @@ bool SshTunnel::connect(const Glib::ustring &server, const Glib::ustring &userna
     if (!verify_host())
         return false;
 
+    ssh_userauth_none(m_ssh, nullptr);
```
Comment (zrax) [inline sshtunnel.cpp]: Should probably check the return code here:
```suggestion
    if (ssh_userauth_none(m_ssh, nullptr) == SSH_AUTH_SUCCESS)
        return true;
```

File: sshtunnel.cpp [hunk 2]
```diff
@@ -363,6 +372,80 @@ bool SshTunnel::prompt_password()
...
+    AppSettings settings;
+
+    int result = ssh_userauth_kbdint(m_ssh, nullptr, nullptr);
+    while (result == SSH_AUTH_INFO) {
+        const char *name, *instruction;
+        int nprompts;
+
+        name = ssh_userauth_kbdint_getname(m_ssh);
+        instruction = ssh_userauth_kbdint_getinstruction(m_ssh);
+        nprompts = ssh_userauth_kbdint_getnprompts(m_ssh);
```
Comment (zrax) [inline sshtunnel.cpp]: Style preference:
```suggestion
        const char *name = ssh_userauth_kbdint_getname(m_ssh);
        const char *instruction = ssh_userauth_kbdint_getinstruction(m_ssh);
        int nprompts = ssh_userauth_kbdint_getnprompts(m_ssh);
```

File: sshtunnel.cpp [hunk 3]
```diff
@@ -363,6 +372,80 @@ bool SshTunnel::prompt_password()
...
+            dialog.hide();
+
+            if (response != Gtk::RESPONSE_OK)
+                return false;
+        }
+
+        for (int i=0; i<nprompts; i++) {
+            Gtk::Entry* input_entry = dynamic_cast<Gtk::Entry*>(grid->get_child_at(1, i + 1));
+            if (input_entry) {
+                ssh_userauth_kbdint_setanswer(m_ssh, i, input_entry->get_text().c_str() );
```
Comment (zrax) [inline sshtunnel.cpp]:
```suggestion
                ssh_userauth_kbdint_setanswer(m_ssh, i, input_entry->get_text().c_str());
```

Comment (pacc) [reply sshtunnel.cpp]: Updated and did some small adjustments:
- assume all auth methods if none are marked as available
- show interactive dialog even with no prompts if there is an instruction

**Comment by pacc** [comment]
Updated and did some small adjustments:

assume all auth methods if none are marked as available
show interactive dialog even with no prompts if there is an instruction

**Review by zrax — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**